src/lib/device/virtual_dev.cpp

```diff
@@ -118,16 +118,18 @@ void VirtualDev::add_chunk(cshared< Chunk >& chunk, bool is_fresh_chunk) {
                                    chunk->physical_dev()->align_size(), chunk->size(), m_auto_recovery,
                                    chunk->chunk_id(), is_fresh_chunk);
     chunk->set_block_allocator(std::move(ba));
-    chunk->set_vdev_ordinal(m_all_chunks.size());
+    // TODO: when vdev_ordinal is  used, revisit here to make sure it is set correctly;
+    chunk->set_vdev_ordinal(m_total_chunk_num++);
     m_pdevs.insert(chunk->physical_dev_mutable());
-    m_all_chunks.push_back(chunk);
+    m_all_chunks[chunk->chunk_id()] = chunk;
     m_chunk_selector->add_chunk(chunk);
 }
 
 void VirtualDev::remove_chunk(cshared< Chunk >& chunk) {
     std::unique_lock lg{m_mgmt_mutex};
     m_all_chunks[chunk->chunk_id()].reset();
     m_all_chunks[chunk->chunk_id()] = nullptr;
+    m_total_chunk_num++;
```

Comment:
Why are we incrementing `m_total_chunk_num` on both add and remove chunk?

Reply:
sorry , this is an mistake, thanks